src/Controller/Async/Stack.php

```diff
@@ -38,21 +41,28 @@ public function add(Request $request)
         $filename = $request->request->get('filename');
 
         $stack = $this->app['stack'];
+        $dispatcher = $this->app['dispatcher'];
 
         /** @var FileInterface|null $removed */
         $file = $stack->add($filename, $removed);
 
         $panel = $this->render('@bolt/components/stack/panel-item.twig', ['file' => $file]);
+        $panelEvent = new GetResponseForControllerResultEvent($this->app['kernel'], $request, HttpKernelInterface::SUB_REQUEST, $panel);
+        $dispatcher->dispatch(KernelEvents::VIEW, $panelEvent);
+
         $list = $this->render('@bolt/components/stack/list-item.twig', ['file' => $file]);
+        $listEvent = new GetResponseForControllerResultEvent($this->app['kernel'], $request, HttpKernelInterface::SUB_REQUEST, $list);
+        $dispatcher->dispatch(KernelEvents::VIEW, $listEvent);
```

Comment:
This feels hacky to me. I think we should either call twig directly here or do a sub request.